packages/react-native-worklets/Common/cpp/worklets/WorkletRuntime/WorkletRuntimeDecorator.cpp

```diff
@@ -173,6 +173,13 @@ void WorkletRuntimeDecorator::decorate(
         return makeShareableInitializer(rt, value.asObject(rt));
       });
 
+  jsi_utils::installJsiFunction(
+      rt,
+      "_makeShareableFunction",
+      [](jsi::Runtime &rt, const jsi::Value &value) {
+        return makeShareableFunction(rt, value.getObject(rt).getFunction(rt));
```

Comment:
Why isn't it `value.asObject(rt).asFunction(rt)`?

```suggestion
        return makeShareableFunction(rt, value.getObject(rt).getFunction(rt));
```

Reply:
`getFunction` does an extra check to see if the value is a function, but as I'm thinking now it doesn't make sense since we're checking that on the JS side